harness/tests/tensorboard/test_s3.py

```diff
@@ -23,12 +23,34 @@
 
 
 @pytest.mark.parametrize("prefix", [None, "my/test/prefix/"])
-def test_s3_build(prefix: Optional[str]) -> None:
+def test_s3_build_sync(prefix: Optional[str]) -> None:
     env = test_util.get_dummy_env()
     conf = copy.deepcopy(default_conf)
     conf["prefix"] = prefix
-    manager = tensorboard.build(env.det_cluster_id, env.det_experiment_id, env.det_trial_id, conf)
+    manager = tensorboard.build(
+        env.det_cluster_id,
+        env.det_experiment_id,
+        env.det_trial_id,
+        conf,
+        async_upload=False,
+    )
+    assert isinstance(manager, tensorboard.S3TensorboardManager)
+
+
+@pytest.mark.parametrize("prefix", [None, "my/test/prefix/"])
+def test_s3_build_async(prefix: Optional[str]) -> None:
```

Comment:
Is this test necessary?  Is there actually a risk that `async_upload` could change the type that is built by `tensorboard.build()`?

Reply:
Agree we can delete this